pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +320,338 @@ func TestMemberService(t *testing.T) {
 		})
 	}
 }
+
+func TestStatefulSetForRack(t *testing.T) {
+	basicRack := scyllav1.RackSpec{
+		Name: "rack",
+		Storage: scyllav1.StorageSpec{
+			Capacity: "1Gi",
+		},
+	}
+	basicSC := &scyllav1.ScyllaCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "basic",
+		},
+		Spec: scyllav1.ScyllaClusterSpec{
+			Datacenter: scyllav1.DatacenterSpec{
+				Name: "dc",
+				Racks: []scyllav1.RackSpec{
+					basicRack,
+				},
+			},
+			ImagePullSecrets: []corev1.LocalObjectReference{
+				{
+					Name: "basic-secrets",
+				},
+			},
+		},
+		Status: scyllav1.ScyllaClusterStatus{
+			Racks: map[string]scyllav1.RackStatus{},
+		},
+	}
+	var existingSts *appsv1.StatefulSet
+	sidecarImage := ""
+
+	matchLabels := naming.RackLabels(basicRack, basicSC)
+	rackLabels := naming.RackLabels(basicRack, basicSC)
+	rackLabels[naming.ScyllaVersionLabel] = basicSC.Spec.Version
+
+	placement := basicRack.Placement
+	if placement == nil {
+		placement = &scyllav1.PlacementSpec{}
+	}
+	opt := true
+
+	storageCapacity, _ := resource.ParseQuantity(basicRack.Storage.Capacity)
+
+	expectedSts := &appsv1.StatefulSet{
```

Comment:
oh this is painful to work with. Do something simillar to this approach: https://github.com/scylladb/scylla-operator/blob/9c745f7aef45c12cee0ad40c8a7989c50421198e/pkg/resourceapply/apps_test.go#L145-L150

Reply:
done